packages/deployments/contracts/contracts/lib/Connext/ConnextLogic.sol

```diff
@@ -864,6 +878,177 @@ library ConnextLogic {
+    // TODO: Should we call approve(0) and approve(totalRepayAmount) instead? or with a try catch to not affect gas on all cases?
+    // Example: https://github.com/aave/aave-v3-periphery/blob/ca184e5278bcbc10d28c3dbbc604041d7cfac50b/contracts/adapters/paraswap/ParaSwapRepayAdapter.sol#L138-L140
+    SafeERC20Upgradeable.safeIncreaseAllowance(IERC20Upgradeable(vars.adopted), _args.aavePool, vars.totalRepayAmount);
+
+    // TODO: What if backUnbacked call fails? For example if the asset is inactive, paused or frozen?
+    // What is the impact of a reconcile to be reverted? Can be re-executed later?
+    IAavePool(_args.aavePool).backUnbacked(vars.adopted, vars.backUnbackedAmount, vars.portalFee);
+
+    // TODO should emit some event with the repayment info?
+
+    // TODO: Do we need to check balance before and after to get the exact amount paid and give the routers the rest?
+    // Aave accounts a global unbacked variable per asset for all, not by address/bridge.
+    // Someone can repay more than it should, so then a the moment of calling backUnbacked()
+    // aave can pull a smaller amount than backUnbackedAmount. So there will be an extra amount of assets that needs to be assigned
+    // See https://github.com/aave/aave-v3-core/blob/feb3f20885c73025f40cc272b59e7eacfaa02fe4/contracts/protocol/libraries/logic/BridgeLogic.sol#L121
+
+    // Account leftovers in adopted to routers
+    return (vars.adoptedAmountOut - vars.totalRepayAmount, vars.adopted);
+  }
+
+  /**
+   * @notice Calculates the amount to be repaid to Aave Portal. If there is no enough amount to repay the unbacked and the fee,
+   * it will partially repay prioritizing the unbacked amount.
+   * @dev Assumes the fee is proportional to the unbackedAmount.
+   * @return The total amount to be repaid
+   * @return The unbacked amount to be backed
+   * @return The fee amount to be paid
+   */
+  function _calculatePortalRepayment(
+    uint256 availableAmount,
+    uint256 portalTransferAmount,
+    uint256 portalFeeNumerator,
+    uint256 portalFeeDenominator
+  )
+    public
+    pure
+    returns (
+      uint256,
+      uint256,
+      uint256
+    )
+  {
+    uint256 portalFee = (portalTransferAmount * portalFeeNumerator) / portalFeeDenominator;
+    uint256 backUnbackedAmount = portalTransferAmount;
+    uint256 totalRepayAmount = backUnbackedAmount + portalFee;
+
+    // If not enough funds to repay the transfer + fees
+    // try to repay as much as unbacked as possible
+    if (totalRepayAmount > availableAmount) {
+      if (availableAmount > backUnbackedAmount) {
+        // Repay the whole transfer and a partial amount of fees
+        portalFee = availableAmount - backUnbackedAmount;
+      } else {
+        // Repay a partial amount of the transfer and no fees
+        backUnbackedAmount = availableAmount;
+        portalFee = 0;
+      }
```

Comment:
will need to make sure we can pay as a router separately from the reconcile flow

also, can you add some comments around the asset that these numbers correspond to?

Reply:
Updated the docs and added a new method `repayAavePortal` to allow a router to manually repay